### PR TITLE
Fix image tags, add edgex-core-consul

### PR DESCRIPTION
**Pull request by ramay1**
Resolves deploy errors due to missing edgex-core-consul and
incorrect image tags.

### CODE REVIEW

File: deployments/consul-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+apiVersion: extensions/v1beta1
```
Comment (rohitsardesai83) [inline deployments/consul-deployment.yaml:1]: are you able to run the edgex services with consul ? I faced errors when I deployed consul . kube-dns was repeatedly exiting. That's why I avoided using consul and instead used kubernetes services for service discovery.Kubernetes does service discovery using kube-dns and kube-proxy , so consul isn't needed .

**Review by rohitsardesai83 — CHANGES_REQUESTED** [review]
are you able to run the edgex services with consul ? I faced errors when I deployed consul . kube-dns was repeatedly exiting. That's why I avoided using consul and instead used kubernetes services for service discovery.Kubernetes does service discovery using kube-dns and kube-proxy , so consul isn't needed

File: deployments/consul-deployment.yaml (hunk 1 above)
Comment (ramay1) [reply deployments/consul-deployment.yaml:1]: Hi, yes; without core-consul, core-command and other services failed.

Comment (rohitsardesai83) [reply deployments/consul-deployment.yaml:1]: can you post the output of the errors ? also can you post the status of all services after running consul ?

Comment (ramay1) [reply deployments/consul-deployment.yaml:1]: vagrant@controller03:~/edgex-on-kubernetes$ kubectl get pods
NAME                                           READY     STATUS    RESTARTS   AGE
edgex-core-command-555bb7c88b-lqnrj            1/1       Running   0          20m
edgex-core-consul-55884b8d68-wbp9d             1/1       Running   0          21m
edgex-core-data-5f794999bd-2ns5c               1/1       Running   0          20m
edgex-core-metadata-5697764b49-q7ksd           1/1       Running   0          20m
edgex-export-client-676758f7cc-hwndm           1/1       Running   0          20m
edgex-export-distro-589b887c56-dgbs8           1/1       Running   0          20m
edgex-mongo-7b789ddcfc-d44wz                   1/1       Running   0          21m
edgex-support-logging-6b6b9bbc-z4pmj           1/1       Running   0          21m
edgex-support-notifications-5f79467d5b-m92cv   1/1       Running   0          21m
edgex-support-scheduler-565b6d6d57-49hb8       1/1       Running   0          20m
rulesengine-576d5c4bbd-6tz9l                   1/1       Running   0          19m
vagrant@controller03:~/edgex-on-kubernetes$

Comment (ramay1) [reply deployments/consul-deployment.yaml:1]: All services running.

Comment (ramay1) [reply deployments/consul-deployment.yaml:1]: vagrant@controller03:~/edgex-on-kubernetes$ curl -i -H GET http://10.233.41.32:48081/api/v1/ping
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Content-Type: text/plain;charset=UTF-8
Content-Length: 4
Date: Thu, 21 Jun 2018 23:23:09 GMT

pongvagrant@controller03:~/edgex-on-kubernetes$

vagrant@controller03:~/edgex-on-kubernetes$ curl -i -H GET http://10.233.33.65:48075/api/v1/ping
HTTP/1.1 200 OK
Server: Apache-Coyote/1.1
Content-Type: text/plain;charset=UTF-8
Content-Length: 4
Date: Thu, 21 Jun 2018 23:26:28 GMT

pongvagrant@controller03:~/edgex-on-kubernetes$

Comment (ramay1) [reply deployments/consul-deployment.yaml:1]: Error after fixing container image tags:

vagrant@controller03:~/edgex-on-kubernetes$ kubectl logs edgex-core-command-658b64ff5-pvbv2
...
...
[2018-06-18 20:01:49.691] boot - 6 ERROR [main] --- ConsulPropertySourceLocator: Fail fast is set and there was an error reading configuration from consul.
[2018-06-18 20:01:49.692] boot - 6 ERROR [main] --- SpringApplication: Application startup failed
com.ecwid.consul.transport.TransportException: java.net.UnknownHostException: edgex-core-consul: Name does not resolve
...
...

Comment (ramay1) [reply deployments/consul-deployment.yaml:1]: vagrant@controller03:~/edgex-on-kubernetes$ kubectl get pods
NAMESPACE     NAME                                           READY     STATUS             RESTARTS   AGE
default       edgex-core-command-658b64ff5-pvbv2             0/1       CrashLoopBackOff   21         1h
default       edgex-core-data-677567fb45-5tzv5               0/1       CrashLoopBackOff   21         1h
default       edgex-core-metadata-d5bb5f5f6-lv8dd            0/1       CrashLoopBackOff   21         1h
default       edgex-export-client-5cc88f4648-5mdzq           0/1       CrashLoopBackOff   21         1h
default       edgex-export-distro-586765c6cc-27wvm           0/1       CrashLoopBackOff   21         1h
default       edgex-mongo-7b789ddcfc-mzzk9                   1/1       Running            0          1h
default       edgex-support-logging-64bb9445c4-q78w5         0/1       CrashLoopBackOff   21         1h
default       edgex-support-notifications-698844c68b-f7qls   0/1       CrashLoopBackOff   21         1h
default       edgex-support-scheduler-648b57496-fns9k        0/1       CrashLoopBackOff   21         1h
default       rulesengine-f6f87fd4f-scw9b                    0/1       CrashLoopBackOff   21         1h

Comment (rohitsardesai83) [reply deployments/consul-deployment.yaml:1]: why do we need consul if we can resolve services using kube-proxy and kube-dns  ?

Comment (rohitsardesai83) [reply deployments/consul-deployment.yaml:1]: I have described about this in my blog : 
https://www.edgexfoundry.org/blog/2018/01/30/huawei-runs-edgex-foundry-kubernetes/

**Pull request merged**